### DISCUSSION TITLE
Salt Edge data not being mapped correctly?

### DISCUSSION THREAD

**Discussion by codeghari** [firefly-iii/firefly-iii#5583] [Firefly III]
Hi hope all is well with the new year.

I have recently setup Firefly-iii and think I can make transfers correctly work with my saltedge import but may need some help as the importer is not detecting the Opposing account (name) correctly when I try to map. Instead it tries to use unknown source/unknown destination instead which are expenses instead of transfers. Is this something that I can change in the firefly importer to correctly pull the transfer information from my saltedge json?

![image](https://user-images.githubusercontent.com/78848307/150619467-23971fbe-4ee5-4f15-9992-e46a5c161447.png)

Here is a sample JSON from saltedge that I assume Firefly-iii importer is using.

{
#this is a unique value per transaction
"id":600000000000000000,

#this is a fixed value designating the source account
"account_id":660000000123456789,

#this is saltedge category
"category":"transfer",

"currency_code":"AUD",
"amount":1.0,
"description":"Transfer from xx0001 CommBank app ",
"extra":{"closing_balance":2.00,

#this is the saltedge defined destination account which is an asset in firefly-iii. this number is the account number of the asset and the name of the asset.
"transfer_account_name":"000-001 1000 0001",
"account_balance_snapshot":2.0,
"categorization_confidence":1}

**Comment by JC5** [comment]
Could you share a few uncensored / uncommented entries (over email if necessary at james@firefly-iii.org)?

**Comment by codeghari** [comment]
Sent you through uncensored JSONs :)

**Comment by taker218** [comment]
I have the same problem.
In the Salt Edge transaction all the information is complete, should I send you the transaction details along with the import log as well? Maybe this could further help in getting down to the problem.

**Reply by JC5** [reply]
Sure thing!

**Reply by taker218** [reply]
Just sent the mail with the files :)

**Comment by JC5** [comment]
Alright so this completely fell of my list, I apologize for the mix-up. If you're still importing using Firefly III (or using the tool at all 😬), this is something you have to fix with rules, unfortunately.

It looks like your bank supplies all details in a single field (the description) and nothing in the payee fields. The data importer will not be able to export that data. You can however, use Firefly III rules to set the destination account based on the content of the description.

I'm sorry you had to wait a bit for a disappointing answer, but at the moment this is the best I can do.

**Reply by taker218** [reply]
Thanks for the answer. Since Salt Edge discontinued the integration of my bank and the new one is not available for Firefly users I switched to the csv import which seems to be working just fine.
